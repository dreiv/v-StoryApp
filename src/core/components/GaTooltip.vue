<script setup lang="ts">
import { PopperWrapper } from 'floating-vue'

defineProps<{ content?: string }>()
</script>

<template>
  <PopperWrapper theme="tooltip">
    <slot />

    <template #popper>
      <header :class="$style.header" v-if="$slots.header">
        <slot name="header" />
      </header>
      <slot name="content">{{ content }}</slot>

      <footer :class="$style.footer" v-if="$slots.footer">
        <slot name="footer" />
      </footer>
    </template>
  </PopperWrapper>
</template>

<style module>
.header {
  margin-bottom: var(--ga-size-4);
  font-weight: var(--font-weight-semibold);
}

.footer {
  display: flex;
  gap: var(--ga-size-spacing-03);
  margin-top: var(--ga-size-16);
}
</style>

<style>
.v-popper--theme-tooltip {
  display: inline-flex;

  .v-popper__inner {
    background-color: var(--ga-color-surface-action-hover);
    padding: var(--ga-size-8);
    color: var(--ga-color-text-on-action);
  }

  .v-popper__arrow-outer {
    border-color: var(--ga-color-surface-action-hover);
  }
}
</style>
