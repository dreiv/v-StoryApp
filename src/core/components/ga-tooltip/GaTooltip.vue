<script setup lang="ts">
import { Tooltip } from 'floating-vue'

defineProps<{ content?: string }>()
</script>

<template>
  <tooltip>
    <slot />

    <template #popper>
      <header :class="$style.header" v-if="$slots.header">
        <slot name="header" />
      </header>
      <slot name="content">{{ content }}</slot>

      <footer :class="$style.footer" v-if="$slots.footer">
        <slot name="footer" />
      </footer>
    </template>
  </tooltip>
</template>

<style module>
.header {
  margin-bottom: var(--ga-size-spacing-02);
  font-weight: var(--ga-font-weight-semibold);
  font-size: var(--ga-text-md-font-size);
  line-height: var(--ga-text-md-line-height);
  font-family: var(--ga-font-family-primary);
  letter-spacing: var(--ga-text-md-tracking);
}

.footer {
  display: flex;
  gap: var(--ga-size-spacing-03);
  margin-top: var(--ga-size-spacing-05);
}
</style>

<style>
.v-popper--theme-tooltip {
  display: inline-flex;
  font-weight: var(--ga-font-weight-normal);
  font-size: var(--ga-text-md-font-size);
  line-height: var(--ga-text-md-line-height);
  font-family: var(--ga-font-family-primary);
  letter-spacing: var(--ga-text-md-tracking);

  .v-popper__inner {
    background-color: var(--ga-color-surface-action-hover);
    padding: var(--ga-size-spacing-03);
    color: var(--ga-color-text-on-action);
  }

  .v-popper__arrow-outer {
    border-color: var(--ga-color-surface-action-hover);
  }
}
</style>
