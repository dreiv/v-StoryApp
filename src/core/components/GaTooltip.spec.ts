import { shallowMount, type VueWrapper } from '@vue/test-utils'
import GaTooltip from './GaTooltip.vue'

vi.mock('floating-vue', () => ({ Tooltip: 'tooltip' }))

describe('Tooltip', () => {
  let wrapper: VueWrapper
  beforeAll(() => {
    wrapper = shallowMount(GaTooltip, { slots: { default: 'mock content' } })
  })

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
