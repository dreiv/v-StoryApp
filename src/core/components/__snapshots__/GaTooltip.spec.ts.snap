// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tooltip > should render correctly 1`] = `"<tooltip>mock content</tooltip>"`;
